test/auto_parallel/hybrid_strategy/parallel_api.py

```diff
@@ -244,34 +236,46 @@ def parallel_model(self, layer, optimizer=None):
-                        PrepareLayerOutput(self.output_transpose),
-                    ],
-                    "llama.layers.*.self_attn.q_proj": [
-                        ColWiseParallel(),
-                        PrepareLayerOutput(self.output_transpose),
-                    ],
-                    "llama.layers.*.self_attn.k_proj": [
-                        ColWiseParallel(),
-                        PrepareLayerOutput(self.output_transpose),
-                    ],
-                    "llama.layers.*.self_attn.v_proj": [
-                        ColWiseParallel(),
-                        PrepareLayerOutput(self.output_transpose),
-                    ],
-                    "llama.layers.*.self_attn.o_proj": [
-                        RowWiseParallel(),
-                        PrepareLayerInput(self.input_transpose),
-                    ],
-                    "llama.layers.*.mlp.gate_proj": ColWiseParallel(),
-                    "llama.layers.*.mlp.up_proj": ColWiseParallel(),
-                    "llama.layers.*.mlp.down_proj": RowWiseParallel(),
-                    "lm_head.weight": ColWiseParallel(),
-                    "lm_head": PrepareLayerInput(self.input_transpose),
-                }
+            else:
+                if self.prepare_input_output:
+                    plan = {
+                        "llama.embed_tokens": ColWiseParallel(),
+                        "llama.layers.*.self_attn.q_proj": ColWiseParallel(),
+                        "llama.layers.*.self_attn.k_proj": ColWiseParallel(),
+                        "llama.layers.*.self_attn.v_proj": ColWiseParallel(),
+                        "llama.layers.*.self_attn.o_proj": RowWiseParallel(),
+                        "llama.layers.*.mlp.gate_proj": ColWiseParallel(),
+                        "llama.layers.*.mlp.up_proj": ColWiseParallel(),
+                        "llama.layers.*.mlp.down_proj": RowWiseParallel(),
+                        "lm_head.weight": ColWiseParallel(),
+                        "llama.layers.*.input_layernorm": SequenceParallel(),
+                        "llama.layers.*.post_attention_layernorm": SequenceParallel(),
+                        "llama.norm": SequenceParallel(),
+                    }
+                else:
+                    plan = {
+                        "llama.embed_tokens": [
+                            ColWiseParallel(),
+                            SequenceParallelBegin(),
+                        ],
+                        "llama.layers.*.self_attn.q_proj": ColWiseParallel(),
+                        "llama.layers.*.self_attn.k_proj": ColWiseParallel(),
+                        "llama.layers.*.self_attn.v_proj": ColWiseParallel(),
+                        "llama.layers.*.self_attn.o_proj": RowWiseParallel(),
+                        "llama.layers.*.self_attn": [
+                            SequenceParallelEnd(),
+                            SequenceParallelBegin(),
+                        ],
```

Comment:
This writing style may not be easy to understand, we need to think about whether there is a more intuitive way to write it

Reply:
Will discuss this later, if find a better way, will update this.